File: 프로그래머스/lv3/43105. 정수 삼각형/정수 삼각형.py
def solution(triangle):
    answer = 0
    dp = [[0 for _ in range(len(triangle)+1)] for _ in range(len(triangle))]
    
    dp[0][1] = triangle[0][0]
    for i in range(1, len(triangle)):
        for j in range(1, i+2):
            dp[i][j] = max(dp[i-1][j-1], dp[i-1][j]) + triangle[i][j-1]
    
    return max(dp[len(triangle)-1])

File: 프로그래머스/lv3/43105. 정수 삼각형/정수 삼각형.java
import java.util.Arrays;

class Solution {
    public int solution(int[][] triangle) {
        int triangleLength = triangle.length;   
		int[][] dp = new int[triangleLength][triangleLength+1];

		dp[0][1] = triangle[0][0];
		for (int i = 1; i < triangleLength; i++) {
			for (int j = 1; j < i+2; j ++) {
				dp[i][j] = Math.max(dp[i - 1][j - 1], dp[i - 1][j]) + triangle[i][j - 1];
			}
		}

		return Arrays.stream(dp[triangleLength-1]).max().getAsInt();
    }
}